fedimint-server/src/config/api.rs

```diff
@@ -0,0 +1,192 @@
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+use async_trait::async_trait;
+use fedimint_core::core::ModuleInstanceId;
+use fedimint_core::db::Database;
+use fedimint_core::module::{
+    api_endpoint, ApiAuth, ApiEndpoint, ApiEndpointContext, ApiError, ApiRequestErased,
+    DecryptionKey,
+};
+use jsonrpsee::server::{ServerBuilder, ServerHandle};
+use jsonrpsee::RpcModule;
+
+use crate::net::api::{attach_endpoints, HasApiContext, RpcHandlerCtx};
+
+/// The config state needed for serving the config API
+pub struct ConfigApi {
```

Comment:
It's more of a `ConfigGenApi`, isn't it?

Reply:
Well technically it will be used after config gen runs as well.  But to address @dpc's point I could prefix these structs with `ConfigApi` to better namespace them.